pandas/_libs/tslibs/conversion.pyx

```diff
@@ -355,6 +363,31 @@ cdef _TSObject convert_datetime_to_tsobject(datetime ts, object tz,
         obj.value += nanos
         obj.dts.ps = nanos * 1000
 
+    obj.fold = 0
+    if tz is not None:
+        if is_utc(tz):
+            pass
+        elif is_tzlocal(tz):
+            pass
+            # TODO: think on how we can infer fold for local Timezone
+            # and adjust value for fold
+        else:
+            trans, deltas, typ = get_dst_info(tz)
+
+            if typ == 'pytz' or typ == 'dateutil':
```

Comment:
Maybe just:

```
if (typ == 'pytz' and fold == 0) or (typ == 'dateutil' and fold == 1):
    pos = trans.searchsorted(obj.value, side='right') - 1
    pos = _adjust_tsobject_for_fold(obj, trans, deltas, pos, fold)
```

Reply:
Thanks, that looks better. In theory, `get_dst_info` can return `typ == 'static'`, but we don't know if that ever happens (there is a TODO in timezones.pyx), and even if it does, it's not like we can infer anything from that.

Reply:
Ah I see. Missed that `pos` was necessary.